sc2/bot_ai.py

```diff
@@ -715,44 +753,38 @@ def _prepare_step(self, state):
             await self.on_building_construction_complete(unit)
 
     async def _issue_unit_dead_events(self):
+        """ Its called when a building or unit is killed"""
         for unit_tag in self.state.dead_units:
             await self.on_unit_destroyed(unit_tag)
 
     async def on_unit_destroyed(self, unit_tag):
-        """ Override this in your bot class. """
-        pass
+        """ Override this in your bot class. Returns the tag of all destroyed units on that step"""
 
     async def on_unit_created(self, unit: Unit):
-        """ Override this in your bot class. """
-        pass
+        """ Override this in your bot class. Returns the Unit() instance of all created units on that step"""
 
     async def on_building_construction_started(self, unit: Unit):
-        """ Override this in your bot class. """
-        pass
+        """ Override this in your bot class. Returns the Unit() instance of all buildings that started on that step """
 
     async def on_building_construction_complete(self, unit: Unit):
-        """ Override this in your bot class. """
-        pass
+        """ Override this in your bot class. Returns the Unit() instance of all buildings that finished on that step """
 
     async def on_upgrade_complete(self, upgrade: UpgradeId):
-        """ Override this in your bot class. """
-        pass
+        """ Override this in your bot class. Returns the UpgradeId of all upgrades that finished on that step """
```

Comment:
same comment as `on_unit_destroyed`

Reply:
I'll keep the docs even on private or protected methods for consistency